src/preciceMap.cpp

```diff
@@ -237,15 +288,28 @@ int main(int argc, char* argv[])
   edgeMap.clear();
   VLOG(1) << "Mesh setup completed on Rank " << MPIrank;
     
-  interface.initialize();
+  auto dt = interface.initialize();
 
   if (interface.isActionRequired(precice::constants::actionWriteInitialData())) {
     VLOG(1) << "Write initial data for participant " << participant;
-    interface.writeBlockScalarData(dataID, mesh.data.size(), vertexIDs.data(), mesh.data.data());
-    std::ostringstream oss;
-    for(size_t i = 0; i<std::min((size_t)10, mesh.data.size()); ++i)
-        oss << mesh.data[i] << ' ';
-    VLOG(1) << "Data written: " << oss.str();
+
+    
+    
+    if (vectordata){
+      size_t i=0;
+      VectorData initialdata = readVectorData(meshes[0]);
+
+      for(auto const& vertexID : vertexIDs){
```

Comment:
As mentioned earlier, we should use the `writeBlockVectorData` variant. The non-block version is more expensive as it has to sanitize the user input for each call.